lib/frontman/context.rb

```diff
@@ -38,28 +38,41 @@ def wrap_layout(layout)
         .returns(T.untyped)
     end
     def content_for(key, content = nil)
-      # Haml is not designed to do handle content_for properly
-      # so we need to hack the buffer of haml that is set
-      # inside the context by haml
-      save_buffer
+      if block_given?
+        content = get_content_buffer(content, &Proc.new)
+      else
+        content = get_content_buffer(content)
+      end
 
-      content ||= ''
+      # We store the the content block inside the current page
+      # because we don't know which renderer/layout/template will need it
+      current_page = Frontman::App.instance.current_page
+      current_page.content_blocks[key.to_sym] = content unless current_page.nil?
+    end
 
+    sig do
+      params(key: T.any(String, Symbol), content: T.untyped)
+        .returns(T.untyped)
+    end
+    def append_content(key, content = nil)
       if block_given?
-        # We don't save the content of the yield, it will be saved in the buffer
-        yield
-
-        # The buffer now contains the content of the yield
-        content = load_buffer
+        content = get_content_buffer(content, &Proc.new)
```

Comment:
Keep in mind we are adding additional overhead by abstracting this method. https://mudge.name/2011/01/26/passing-blocks-in-ruby-without-block/ I chose the most performant method